src/atoms/input.jsx

```diff
@@ -324,88 +252,232 @@ class Input extends React.PureComponent {
                 onChange(newValue);
             }
 
-            this._shouldShowRequiredIndicator(newValue);
+            this.shouldShowRequiredIndicator(newValue);
         }
     }
 
-    _onNumberToggleDownClick(event) {
-        this._onNumberToggleClick('down');
+    onNumberToggleDownClick() {
+        this.onNumberToggleClick('down');
     }
 
-    _onNumberToggleUpClick(event) {
-        this._onNumberToggleClick('up');
+    onNumberToggleUpClick() {
+        this.onNumberToggleClick('up');
     }
 
-    _shouldShowRequiredIndicator(value) {
+    getType() {
+        const {
+            type,
+        } = this.props;
+        let newType;
+
+        switch (type) {
+            case 'email':
+                newType = 'email';
+                break;
+            case 'number':
+                newType = 'number';
+                break;
+            case 'password':
+                newType = 'password';
+                break;
+            case 'telephone':
+            case 'tel':
+            case 'phone':
+                newType = 'tel';
+                break;
+            case 'text':
+            default:
+                newType = 'text';
+                break;
+        }
+
+        return newType;
+    }
```

Comment:
Seems to me that, in theory, we shouldn't really need this, unless consumers are passing invalid `type` props.

Reply:
Should remove? Is it a blocker?

Reply:
Nah ... definitely a NAB.  Up to you whether to remove now (and possibly incur small, random regressions if consumers are using non-standard values).  Maybe safer to keep it.  I imagine at some point there was a good reason for it.  Those React CM UI consumers can be sloppy coders sometimes! 😁